config/v1/types.go

```diff
@@ -401,16 +401,20 @@ const (
 
 // IBMCloudServiceName contains a value specifying the name of an IBM Cloud Service,
 // which are used by MAPI, CIRO, CIO, Installer, etc.
-// +kubebuilder:validation:Enum=CIS;COS;DNSServices;GlobalSearch;GlobalTagging;HyperProtect;IAM;KeyProtect;ResourceController;ResourceManager;VPC
+// +kubebuilder:validation:Enum=CIS;COS;COSConfig;DNSServices;GlobalCatalog;GlobalSearch;GlobalTagging;HyperProtect;IAM;KeyProtect;ResourceController;ResourceManager;VPC
```

Comment:
Is this feature being implemented behind a feature gate? /Is it possible to do so?

We can gate the enum values so that they are only available in clusters specifying the appropriate gates now, so if this is linked to an already gated feature we should make sure to include that  here

Reply:
The Endpoint Override support wasn't feature gated originally (GA'd in 4.15).
As we don't currently support reconfigurable serviceEndpoints (post cluster create), these new values would only be validated/used at cluster creation.

However, these new values could be made available (via api update in installer), and if the user attempts to use them, the installer will error out if it hasn't been updated to handle them (as the [installer checks the service names](https://github.com/openshift/installer/blob/f1890771b0a1a40ca15318cfe28e58360ae2e6bd/pkg/tfvars/ibmcloud/ibmcloud.go#L241-L242)).

So, would a feature gate help deal with that situation (where api is updated, but installer isn't able to handle the new names yet)? Although that will only break, if those new names get used in `install-config`.

I do wish to cherry-pick all of these related fixes back to 4.15, so I am interested in knowing how that would affect using a (new) feature gate too.

Reply:
Hmm no, since this feature is only available at install time, I don't think adding a feature gate here would add much value. Especially since we are just extending an existing API.

Uses are not expected to update the infrastructure platform status so since this is only used there, I think getting this merged and then updated in the installer is clean enough

Reply:
Okay, great. Thank you!